### ISSUE TITLE
Allow named exports in js/ts file with autoimport

### ISSUE THREAD

**Issue by huang-julien** [nuxt/nuxt#25340]
### Describe the feature

https://stackblitz.com/edit/nuxt-starter-frkyxu?file=pages%2Findex.vue,components%2FHello.ts,app.vue

not sure if it's a missing feature; We currently cannot import named exports with component dir.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by manniL** [comment]
True! IMO we should handle it as with composables when it is a `.ts` file in components. 👍🏻

**Comment by HigherOrderLogic** [comment]
Hi, I would like to take on this issue. One thing I'm not sure about is how I should name the components with named exports? I'm thinking about `<path><file name><export name>`, or should I just go with `<path><export name>`?

**Comment by huang-julien** [comment]
We should keep the same naming system just like for named export in utils or composables.

The catch here is that we probably need to read the file and add all exports as components with `addComponent` which contains an `export` field.

The other issue is that I'm pretty sure some people export types or non-component varaibles from theses files so we need to know what to do with it.

**Comment by guaijie** [comment]
> True! IMO we should handle it as with composables when it is a `.ts` file in components. 👍🏻



> True! IMO we should handle it as with composables when it is a `.ts` file in components. 👍🏻

![Image](https://github.com/user-attachments/assets/ebfc9b64-3907-4ed3-87b4-34276d7d8ab0)

This is a temporary solution
